### PR TITLE
meson.build fixes to build msvc version

### PR DESCRIPTION
**Pull request by skuater**
- Edited all meson files to add include directory for msvc build
- Edited some .c files

### CODE REVIEW

File: binr/ragg2/ragg2.c [hunk 1]
```diff
@@ -96,7 +96,11 @@ static int openfile (const char *f, int x) {
 #if __UNIX__
 	if (x) fchmod (fd, 0755);
 #endif
+#if _MSC_VER
```
Comment (radare) [inline binr/ragg2/ragg2.c]: uhm, maybe we can add a r_file_truncate() in libr/util/file.c

File: libr/bin/pdb/pdb_downloader.c [hunk 2]
```diff
@@ -106,7 +106,7 @@ void init_pdb_downloader(SPDBDownloaderOpt *opt, SPDBDownloader *pd) {
 	pd->download = download;
 }
 
-void deinit_pdb_downloader(SPDBDownloader *pd) {
+R_API void deinit_pdb_downloader(SPDBDownloader *pd) {
```
Comment (radare) [inline libr/bin/pdb/pdb_downloader.c]: symbols not prefixed with r_ should be public api and therefor dont use the R_API thing

File: libr/bin/pdb/pdb_downloader.c [hunk 3]
```diff
@@ -106,7 +106,7 @@ void init_pdb_downloader(SPDBDownloaderOpt *opt, SPDBDownloader *pd) {
...
 	R_FREE(pd->opt->dbg_file);
```
Comment (radare) [inline libr/bin/pdb/pdb_downloader.c]: missing spaces before (

File: libr/bin/pdb/pdb_downloader.h [hunk 4]
```diff
@@ -25,13 +25,13 @@ typedef struct SPDBDownloader {
 /// \param opt PDB options
 /// \param pdb_downloader PDB downloader that will be init
 ///
-void init_pdb_downloader(SPDBDownloaderOpt *opt, SPDBDownloader *pdb_downloader);
+R_API void init_pdb_downloader(SPDBDownloaderOpt *opt, SPDBDownloader *pdb_downloader);
```
Comment (radare) [inline libr/bin/pdb/pdb_downloader.h]: dont R_API stuff that is not starting by r_

File: shlr/spp/spp.h [hunk 5]
```diff
@@ -12,7 +12,7 @@
 int r_sys_setenv(const char *key, const char *value);
 #endif
 
-#ifdef __WINDOWS__
+#if defined (_MSC_VER) || (__WINDOWS__)
```
Comment (radare) [inline shlr/spp/spp.h]: imho _MSC_VER should be implicit by __WINDOWS__ msvc cant compile for any other platform and i dont think it will ever do

Comment (radare) [inline shlr/spp/spp.h]: also this patch should go into spp, not here

File: shlr/spp/p/spp.h [hunk 6]
```diff
@@ -1,7 +1,6 @@
 /* Mini MCMS :: renamed to 'spp'? */
 
 #include <unistd.h>
-
```
Comment (radare) [inline shlr/spp/p/spp.h]: dont rm this line

File: binr/ragg2/ragg2.c (hunk 1 above)
Comment (skuater) [inline binr/ragg2/ragg2.c]: maybe, but at moment only fews truncate in all r2 code, i think we can leave here

File: shlr/spp/spp.h (hunk 5 above)
Comment (XVilka) [inline shlr/spp/spp.h]: @radare - no, __WINDOWS__ is defined also for Mingw32, Cygwin and MSYS2 (Mingw-w64), while _MSC_VER is defined only for MSVC.

**Comment by radare** [comment]
yep i know. thats why i say that in r_types_base.h we should make __WINDOWS__ be 1 if __MSC_VER is set

**Pull request merged**